module/leftPanel/leftPanelModule.js

```diff
@@ -53,12 +55,29 @@ var LeftPanelModule = class LeftPanelModule {
         this.subModules.forEach(subModule => {
             subModule.enable();
         });
+
+        this.themeSettings = getSettings('theme');
+        this.showSettingsButton = this.themeSettings.get_boolean('show-settings-button-on-panel');
+        this.settingsButtonSetting = this.themeSettings.connect('changed::show-settings-button-on-panel', schema => {
+            this.settingsButtonSetting = schema.get_boolean('show-settings-button-on-panel');
+            this.setSettingsButtonToSettingsButtonSetting(this.settingsButtonSetting);
+        });
+        this.setSettingsButtonToSettingsButtonSetting(this.settingsButtonSetting);
+    }
+
+    setSettingsButtonToSettingsButtonSetting(settingsButtonSetting) { // did I mention settings?
+        if (settingsButtonSetting) {
+            this.settingButtonSubModule.enable();
+        } else {
+            this.settingButtonSubModule.disable();
+        }
```

Comment:
Can you do all of this inside the settingButtonSubModule? In the enable function you check if the settings is true before enabled and you disable only if it has been enable already?

Reply:
Done.